src/components/chartRepo/ChartRepo.tsx

```diff
@@ -152,15 +154,34 @@ function ChartForm({ id = null, name = "", active = false, url = "", authMode =
     const customHandleChange = e => setCustomState(state => ({ ...state, [e.target.name]: { value: e.target.value, error: "" } }))
     const [deleting, setDeleting] = useState(false);
     const [confirmation, toggleConfirmation] = useState(false);
+    const [chartRepoType, setChartRepoType] = useState<string>('PUBLIC')
+    
+    if(chartRepoType==='PUBLIC'){
+        state.auth.value= 'ANONYMOUS'
+    }else{
+        state.auth.value= 'USERNAME_PASSWORD'
```

Comment:
Please use constant for  USERNAME_PASSWORD, ANONYMOUS, PUBLIC

Reply:
done